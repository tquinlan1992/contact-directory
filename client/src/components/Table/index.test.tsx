import React from "react";
import Enzyme, { render } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Table, TableBody, TableHeaders, TableRow } from ".";
Enzyme.configure({ adapter: new Adapter() });

describe("Table", () => {
  describe("with header, body, rows", () => {
    it("renders all the children", () => {
      const wrapper = render(
        <Table>
          <TableHeaders headers={["name", "email", "address"]} />
          <TableBody>
            <TableRow columns={["row_name", "row_email", "row_address"]} />
          </TableBody>
        </Table>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
