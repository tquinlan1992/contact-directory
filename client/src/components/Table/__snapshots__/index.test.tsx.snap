// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table with header, body, rows renders all the children 1`] = `
<table>
  <thead>
    <tr>
      <th>
        name
      </th>
      <th>
        email
      </th>
      <th>
        address
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        row_name
      </td>
      <td>
        row_email
      </td>
      <td>
        row_address
      </td>
    </tr>
  </tbody>
</table>
`;
